src/config.rs

```diff
@@ -1007,6 +1007,65 @@ mod tests {
         Ok(())
     }
 
+    /// Sync message mustn't be sent if self-{status,avatar} is changed by a self-sent message.
```

Comment:
It's checked implicitly. The reason is that `TestContext::pop_sent_msg_opt()` returns None only after timing out, so the test checks that the right sync message is sent instead of that some message isn't sent.

Even if sync messages are sent immediately now, i'm not sure it's a good thing to rely on in tests.